src/Module/README.md

```diff
@@ -3,8 +3,29 @@
 The Module namespace contains the different modules of Friendica.
 Each module is loaded through the [`App`](https://github.com/friendica/friendica/blob/develop/src/App.php).
 
-Rules for Modules:
-- 	Named like the call (i.e. https://friendica.test/contact => `Contact`)
--	Start with capitals and are **not** camelCased.
--	Directly interacting with a given request (POST or GET)
--	Extending [`BaseModule`](https://github.com/friendica/friendica/blob/develop/src/BaseModule.php).
\ No newline at end of file
+There are mainly two types of modules:
+-	frontend modules to interact with users
+-	backend modules to interact with machine requests
+
+### Frontend modules
+
+This type of modules mainly needs a template, which are generally located at
+[view/templates/](https://github.com/friendica/friendica/tree/develop/view/templates).
+
+A frontend module should extend the [`BaseModule`](https://github.com/friendica/friendica/blob/develop/src/BaseModule.php), especially the `content()` method. 
+
+### Backend modules
+
+This type of modules mainly responds either with encoded `XML` or with `JSON` output.
+It isn't intended to respond with human readable text.
+
+A frontend module should extend the [`BaseModule`](https://github.com/friendica/friendica/blob/develop/src/BaseModule.php), especially the `rawContent()` method.
+ 
+### Routing
+
+Every module needs to be accessed within a route.
+The routing mechanism is using a PSR-7 based routing and the routes are defined inside [`Router->collectRoutes()`](https://github.com/friendica/friendica/blob/develop/src/App/Router.php).
```

Comment:
The Router is homemade and directly based on FastRoute: https://github.com/nikic/FastRoute

I didn't check any PSR compatibility.

Reply:
sorry, I checked for "fastroute" in my chronic and found accidentally the followup project ^^